resource/index_bench_test.go

```diff
@@ -9,12 +9,48 @@ import (
 
 func BenchmarkGetResource(b *testing.B) {
 	index := NewIndex()
-	for i := 0; i < 1000; i++ {
-		index.Bind(fmt.Sprintf("route%d", i), schema.Schema{}, nil, DefaultConf)
+	for i := 1; i <= 1000; i++ {
+		index.Bind(fmt.Sprintf("route%04d", i), schema.Schema{}, nil, DefaultConf)
 	}
-	b.ResetTimer()
-	b.ReportAllocs()
-	for i := 0; i < b.N; i++ {
-		index.GetResource("route80", nil)
+	tests := map[string]string{
+		"1st":      "route0001",
+		"100th":    "route0100",
+		"1000th":   "route1000",
+		"NotFound": "notfound",
 	}
+	for name, rsrc := range tests {
+		b.Run(name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				index.GetResource(rsrc, nil)
```

Comment:
No changes needed, this is just a FYI about a sub-test gatcha that applies to parallel/concurrent execution, in case you were not aware.

Using `rsrc` variable here _only_ works as long as the benchmarks here are not run in parallel.

If you where to add `b.Parallel()` above this line, the `rscrc` variables would be _shared_ among all the goroutines, i.e. it's reused for the next iteration of the for loop, leading to a data race condition. `name` is also reused in the next iteration, but as it's used only as an argument to Run, it happens before any new goroutine is started, so it's not a problem. To solve this in the case of parallelism we would need to capture the variable, e.g. like this:

```go
for name := range tests {
        rsrc := tests[name]  // the variable is declared within the iteration, and thus not reused on the next iteration.
```

Extract from https://blog.golang.org/subtests:
```go
func TestGroupedParallel(t *testing.T) {
    for _, tc := range testCases {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) {
```
Again, there are no changes needed for this case, just something to be aware of.